charmander/factories/order_event_factory.py

```diff
@@ -23,3 +24,20 @@ def __init__(self, graph):
                 version="v1",
             ),
         )
+
+    def create_event(self, event_info,  **kwargs):
+        super().create_event(event_info, **kwargs)
+        if event_info.event_type == OrderEventType.OrderDeliveryDetailsAdded:
```

Comment:
This is a bit of a two example, but this is an example of the sort of special logic you might need to add on event creation. The factory gives one, unambiguous point to put this rather than hiding such logic across different controllers which may also need to create `OrderEvent` objects.